scipy/stats/_continuous_distns.py

```diff
@@ -2900,6 +2900,186 @@ def _entropy(self, c):
 genhalflogistic = genhalflogistic_gen(a=0.0, name='genhalflogistic')
 
 
+class genhyperbolic_gen(rv_continuous):
+    r"""A generalized hyperbolic continuous random variable.
+
+    %(before_notes)s
+
+    See Also
+    --------
+    t, norminvgaus, geninvgauss, laplace, cauchy
+
+    Notes
+    -----
+    The probability density function for `genhyperbolic` is:
+
+    .. math::
+
+        f(x, \lambda, \hat{\alpha}, \hat{\beta)} =
+            \frac{(\hat{\alpha}^2 - \hat{\beta}^2)^{\lambda/2}}
+            {\sqrt{2\pi}\hat{\alpha}^{\lambda-0.5}
+            K_\lambda\Big(\sqrt{\hat{\alpha}^2 - \hat{\beta}^2}\Big)}
+            e^{\hat{\beta}(x)} \times \frac{K_{\lambda - 1/2}
+            (\hat{\alpha} \sqrt{1 + x^2})}
+            {(\sqrt{1 + x^2})^{1/2 - \lambda}}
+
+    for :math:`x, \lambda \in \mathbb{R}`,
+    :math:`|\hat{\beta}| \lt \hat{\alpha}` if :math:`\lambda \ge 0`,
+    :math:`|\hat{\beta}| \lt \hat{\alpha}` if :math:`\lambda = 0`,
+    :math:`|\hat{\beta}| \leq \hat{\alpha}` if :math:`\lambda \lt 0`.
+    :math:`K_{\lambda}(.)` denotes the modified Bessel function of the second
+    kind and order :math:`\lambda` (`scipy.special.kn`)
+
+    `genhyperbolic` takes ``p`` as a tail parameter for :math:`\lambda`,
+    ``a`` as a shape parameter for :math:`\hat{\alpha}`,
+    ``b`` as a skewness parameter for :math:`\hat{\beta}`,
+
+    Moments are based on [3]_.
```

Comment:
Looks like this note might belong somewhere else.

Reply:
I think this should go somewhere else. Maybe the end, after [1] and [2] have been mentioned?

Reply:
ah ok i see what you mean. Will try to sneak in [1] and [2] before that